ballerina/tests/constraint_annotation_test.bal

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License
+
+import ballerina/constraint;
+import ballerina/test;
+import ballerina/time;
+
+public type ValidationPerson record {|
```

Comment:
Shall we extend this record to have nested fields? And lets use the `ArrayConstraints` annotation as well.

Reply:
Updated https://github.com/ballerina-platform/module-ballerina-data.xmldata/pull/21/commits/2aae4555d071138dc344efdc2c3f9be682b23e6a